### PR TITLE
07325 pcli clean

### LINKED ISSUE #7325: Implement PCLI Clean Command for platform files

### PR DESCRIPTION
**Pull request by lpetrovic05**
closes #7325

### CODE REVIEW

**Review by alittley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/CleanCommand.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    private Path sdkPath;
+
+    /** Set the path to the sdk directory */
+    @CommandLine.Parameters(description = "the path to the sdk directory")
+    private void setSdkPath(final Path sdkPath) {
+        this.sdkPath = dirMustExist(sdkPath.toAbsolutePath());
+    }
+
+    @Override
+    public Integer call() throws Exception {
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/CleanCommand.java:48]: Is the return value `@Nullable` or `@NonNull`?

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/FileUtils.java [hunk 2]
```diff
@@ -289,4 +290,32 @@ public static void writeAndFlush(final Path file, final IOConsumer<MerkleDataOut
...
     }
+
+    /**
+     * Find files in the given directory that end with the given suffix.
+     * @param dir the directory to search
+     * @param suffix the suffix to match
+     * @return a list of paths to files that match the given suffix
+     * @throws IOException if there is a problem walking the directory
+     */
+    public static List<Path> findFiles(final Path dir, final String suffix) throws IOException {
```
[the remark this answers is not in the document]
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/FileUtils.java:301]: return type from method also needs the marking.

File: platform-sdk/swirlds-cli/src/main/java/com/swirlds/cli/utility/ParameterizedClass.java [hunk 3]
```diff
@@ -61,6 +61,20 @@ protected Path pathMustExist(final Path path) {
         return path;
     }
 
+    /**
+     * Ensure that a path from the command line exists and that it is a directory.
+     *
+     * @param path a path from the command line
+     * @return the path if it passes validation
+     */
+    protected Path dirMustExist(final Path path) {
```
[the remark this answers is not in the document]
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-cli/src/main/java/com/swirlds/cli/utility/ParameterizedClass.java:70]: return type also needs the marking.

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/CleanCommand.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import com.swirlds.platform.config.AddressBookConfig;
+import com.swirlds.platform.util.BootstrapUtils;
+import java.nio.file.Path;
+import java.util.List;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+        name = "clean",
+        mixinStandardHelpOptions = true,
+        description = "Delete all files generated by running the platform.")
```
Comment (poulok) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/CleanCommand.java:35]: Does this also clean up all merkledb files?

**Review by hendrikebbers — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/CleanCommand.java (hunk 4 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/CleanCommand.java:35]: they are saved in the `saved` directory as far as i know, so yes

File: platform-sdk/swirlds-cli/src/main/java/com/swirlds/cli/utility/ParameterizedClass.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (lpetrovic05) [reply platform-sdk/swirlds-cli/src/main/java/com/swirlds/cli/utility/ParameterizedClass.java:70]: done the other methods as well

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/FileUtils.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (lpetrovic05) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/FileUtils.java:301]: done

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/FileUtils.java [hunk 5]
```diff
@@ -289,4 +290,32 @@ public static void writeAndFlush(final Path file, final IOConsumer<MerkleDataOut
...
+        }
+    }
+
+    /**
+     * Delete files in the given directory that end with the given suffix.
+     * @param dir the directory to search
+     * @param suffix the suffix to match
+     * @throws IOException if there is a problem walking the directory or deleting
+     */
+    public static void deleteFiles(final Path dir, final String suffix) throws IOException {
```
[the remark this answers is not in the document]
Comment (lpetrovic05) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/utility/FileUtils.java:315]: done

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/CleanCommand.java (hunk 1 above)
Comment (lpetrovic05) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/CleanCommand.java:48]: no idea, I think @cody-littley wrote the abstract method

**Review by poulok — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Pull request merged**